cardano-node/chairman/Cardano/Chairman.hs

```diff
@@ -83,23 +78,27 @@ import           Cardano.Config.Types (SocketPath(..))
 -- that failures can be detected as early as possible. The progress condition
 -- is only checked at the end.
 --
-chairmanTest :: forall blk. RunNode blk
-             => Tracer IO String
-             -> Protocol IO blk (BlockProtocol blk)
+chairmanTest :: Tracer IO String
+             -> SlotLength
+             -> SecurityParam
              -> DiffTime
              -> Maybe BlockNo
              -> [SocketPath]
+             -> SomeNodeClientProtocol
+             -> NetworkMagic
              -> IO ()
-chairmanTest tracer ptcl runningTime optionalProgressThreshold socketPaths = do
+chairmanTest tracer slotLength securityParam runningTime optionalProgressThreshold socketPaths someNodeClientProtocol nw = do
 
     traceWith tracer ("Will observe nodes for " ++ show runningTime)
     traceWith tracer ("Will require chain growth of " ++ show progressThreshold)
 
+    SomeNodeClientProtocol (p :: ProtocolClient blk (BlockProtocol blk)) <- return $ someNodeClientProtocol
+
```

Comment:
I think it'd be cleaner to do it like the original code did, with a `ptcl` arg of type `forall blk. RunNode blk => ProtocolClient blk (BlockProtocol blk)`. There's no need for the existential type wrapper.